tests/data/test_module.py

```diff
@@ -10,36 +11,83 @@
 
 @pytest.mark.skipif(GITHUB_ACTIONS, reason="only works locally")
 class TestModuleInstantiation:
-    def test_datamodule_initialization(self):
-        temp_data_args = ClimateDatasetArgs(variables=["2m_temperature"], split="train")
-        temp_highres_data_args = ClimateDatasetArgs(
-            variables=["2m_temperature"], split="train"
-        )
-        temp_task_args = DownscalingArgs(
-            temp_data_args,
-            temp_highres_data_args,
-            in_vars=["2m_temperature"],
-            out_vars=["2m_temperature"],
-        )
-        DataModuleArgs(temp_task_args, 2014, 2016, 2017)
-
-    def test_initialization(self):
-        temp_data_args = ERA5Args(
+    def test_map_initialization(self):
+        climate_dataset_args = ERA5Args(
             root_dir=os.path.join(DATA_PATH, "era5/5.625deg/"),
-            variables=["2m_temperature"],
-            years=range(2010, 2015),
+            variables=[
+                "2m_temperature",
+                "geopotential",
+                "temperature_250",
+                "orography",
+                "land_sea_mask",
+            ],
+            years=range(2009, 2015),
             split="train",
         )
-        temp_highres_data_args = ERA5Args(
+        high_res_climate_dataset_args = ERA5Args(
             root_dir=os.path.join(DATA_PATH, "era5/2.8125deg/"),
-            variables=["2m_temperature"],
-            years=range(2010, 2015),
+            variables=[
+                "geopotential",
+                "temperature_250",
+            ],
+            years=range(2009, 2015),
+            split="train",
```

Comment:
If I understand this correctly, this means that for every split, the user has to do this entire process again? I think we can design this better. Option 1: all the splits are created with just one set of arguments. Option 2: we give the user some sort of method for copying an existing `ERA5Args` instance and just changing the years/split. For example,

```python
train_args = ERA5Args(..., years=range(2009, 2015), split="train")
val_args = train_args.copy(years=range(2015, 2017), split="val")
```

Reply:
Currently all splits are created with just one set of arguments. See `DataModuleArgs`, it creates copy of them and based on the other args like `train_start_year`, `val_start_year`, `test_start_year`, etc it automatically changes the appropriate variables for the copies. See my comment on why we need `StackedClimateDatasetArgs` class to understand how this is happening.

Reply:
I have implemented something similar to what you suggested. See `create_copy()` functions of different Args classes to understand how it works. You can also find an example of it under [`test_module.py`](https://github.com/aditya-grover/climate-learn/blob/2f3db813b98ccb91b14709da7796734e5bdba99e/tests/data/test_module.py#L109).